### PR TITLE
add M_TOO_LARGE error code for uploading a too large file

### LINKED ISSUE #6109: Uploading a too large file: wrong error code
When uploading a too large file, synapse responds with a 413 (OK) but with errcode M_UNKNOWN. According to the [spec](https://matrix.org/docs/spec/client_server/latest#post-matrix-media-r0-upload), it should be "M_TOO_LARGE"

Received responseCode: 413
Received responseBody:
`{"errcode":"M_UNKNOWN","error":"Upload request body is too large"}`

### PR DESCRIPTION
**Pull request by AnshulAngaria**
### Pull Request Checklist

Fixes #6109 
Signed-off-by: Anshul Angaria <angaria.anshul37@gmail.com>

-  [x] Pull request is based on the develop branch
-  [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
- [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

File: synapse/rest/media/v1/upload_resource.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 from twisted.web.server import NOT_DONE_YET
 
-from synapse.api.errors import SynapseError
+from synapse.api.errors import SynapseError , Codes
```
Comment (anoadragon453) [inline synapse/rest/media/v1/upload_resource.py]:
```suggestion
from synapse.api.errors import Codes, SynapseError
```

File: changelog.d/6109.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+fix bug when uploading a large file synapse responds with M_UNKNOWN while it should be M_TOO_LARGE according to spec. Contributed by Anshul Angaria
```
Comment (anoadragon453) [inline changelog.d/6109.bugfix]:
```suggestion
Fix bug when uploading a large file synapse responds with M_UNKNOWN while it should be M_TOO_LARGE according to spec. Contributed by Anshul Angaria.
```

Changelog entries must always end in punctuation :)

Comment (anoadragon453) [inline changelog.d/6109.bugfix]:
```suggestion
Fix bug when uploading a large file synapse responds with M_UNKNOWN while it should be M_TOO_LARGE according to spec. Contributed by Anshul Angaria.
```

Changelog entries must always end in punctuation :)

File: synapse/rest/media/v1/upload_resource.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def render_OPTIONS(self, request):
         if content_length is None:
             raise SynapseError(msg="Request must specify a Content-Length", code=400)
         if int(content_length) > self.max_upload_size:
-            raise SynapseError(msg="Upload request body is too large", code=413)
+            raise SynapseError(msg="Upload request body is too large", code=413,  errcode=Codes.TOO_LARGE)
```
Comment (babolivier) [inline synapse/rest/media/v1/upload_resource.py]:
```suggestion
            raise SynapseError(
                msg="Upload request body is too large",
                code=413,
                errcode=Codes.TOO_LARGE,
            )
```

Linter's complaining again :confused:

**Comment by AnshulAngaria** [comment]
Looking forward to contribute further!

File: changelog.d/6109.bugfix [hunk 4]
```diff
@@ -0,0 +1 @@
+Fix bug when uploading a large file synapse responds with M_UNKNOWN while it should be M_TOO_LARGE according to spec. Contributed by Anshul Angaria.
```
Comment (anoadragon453) [inline changelog.d/6109.bugfix]:
```suggestion
Fix bug when uploading a large file: Synapse responds with `M_UNKNOWN` while it should be `M_TOO_LARGE` according to spec. Contributed by Anshul Angaria.
```

**Pull request merged**